core/src/validate.rs

```diff
@@ -177,8 +178,10 @@ fn validate_writable(
 ) {
     // Only User registers are writable, all others cause an error.
     match reg_ref_node {
-        Node(RegisterRef::User(_), _) => {}
-        Node(_, span) => errors.push((CompileError::UnwritableRegister, *span)),
+        Node(RegisterRef::Null | RegisterRef::User(_), _) => {}
+        Node(RegisterRef::InputLength | RegisterRef::StackLength(_), span) => {
```

Comment:
This is the new `or_patterns` feature. It lets you use the `|` operator deep inside patterns, instead of just at the top level.